tests/scripts/rm_hpo_api_tests.sh

```diff
@@ -0,0 +1,350 @@
+	# Example curl command: curl -H "Content-Type: application/json" -d {"id" : "a123", "url" : "http://localhost:8080/searchSpace", "operation" : "EXP_TRIAL_GENERATE_NEW"}  http://localhost:8085/experiment_trials -w n%{http_code}
+	post_experiment_cmd="${content} -d ${json_array_}  http://localhost:8085/experiment_trials -w '\n%{http_code}'"
+	
+	echo "" | tee -a ${LOG_} ${LOG}
+	echo "Command used to post the experiment= ${post_experiment_cmd}" | tee -a ${LOG_} ${LOG}
+	echo "" | tee -a ${LOG_} ${LOG}
+	
+	echo "${post_cmd}" >> ${LOG_} ${LOG}
+	
+	http_code=$(tail -n1 <<< "${post_cmd}")
+	response=$(echo -e "${post_cmd}" | tail -2 | head -1)
+	
+	echo "Response is ${response}" >> ${LOG_} ${LOG}
+}
+
+# Compare the actual result with the expected result
+# input: Test name, expected result 
+function compare_result() {
+	flag=0
+	__test__=$1
+	expected_result=$2
+	expected_behaviour=$3
+	
+	if [[ ! ${actual_result} =~ ${expected_result} ]]; then
+		flag=1
+	fi
+	
+	display_result "${expected_behaviour}" ${__test__} ${flag}
+}
+
+function check_server_status() {
+	searchspace_log_msg="Starting server at http://localhost:8080"
+	service_log_msg="Starting server at http://localhost:8085"
+	
+	echo ""
+	if grep -q "${searchspace_log_msg}" "${TESTS_}/searchspace.log" ; then
+		echo "Searchspace service started successfully..." | tee -a ${LOG_} ${LOG}
+	else
+		echo "Error Starting the searchspace service..." | tee -a ${LOG_} ${LOG}
+		echo "See ${TESTS_}/searchspace.log for more details" | tee -a ${LOG_} ${LOG}
+		exit 0
+	fi
+	
+	if grep -q "${service_log_msg}" "${TESTS_}/service.log" ; then
+		echo "HPO REST API service started successfully..." | tee -a ${LOG_} ${LOG}
+	else
+		echo "Error Starting the HPO REST API service..." | tee -a ${LOG_} ${LOG}
+		echo "See ${TESTS_}/service.log for more details" | tee -a ${LOG_} ${LOG}
+		exit 0
+	fi
+}
+
+# Do a post with JSON array having invalid fields
+# input: Test name
+# output: get the JSON, post it to HPO and compare the result
+function invalid_post(){
+	__test_name__=$1
+	for post_test in "${invalid_post_tests[@]}"
+	do
+		TESTS_="${TEST_DIR}/${post_test}"
+		mkdir -p ${TESTS_}
+		LOG_="${TEST_DIR}/${post_test}.log"
+		
+		echo "************************************* ${post_test} Test ****************************************" | tee -a ${LOG_} ${LOG}
+		echo "" | tee -a ${LOG_} ${LOG}
+
+		case "${post_test}" in
+			invalid-searchspace)
+				exp="valid-experiment"
+				sed 's/"sla_class": "response_time"/"sla_class": "xyz"/g' ${SEARCH_SPACE_JSON} > ${TESTS_}/invalid_searchspace.json
+				echo "Searchspace JSON" | tee -a ${LOG_} ${LOG}
+				cat ${TESTS_}/invalid_searchspace.json | tee -a ${LOG_} ${LOG}
+				echo "" | tee -a ${LOG_} ${LOG}
+				# Start the HPO servers
+				${SCRIPTS_DIR}/start_python_servers.sh -p ${TESTS_} -j ${TESTS_}/invalid_searchspace.json | tee -a ${LOG_} ${LOG}
```

Comment:
I have now renamed start_python_servers.sh to start_hpo_servers.sh, please make this change in this script too.